### DISCUSSION TITLE
Consider adding JSDoc to utility functions in mantine.

### DISCUSSION THREAD

**Discussion by UltraInstinct05** [mantinedev/mantine#604] [Feedback]
`Mantine` has a good number of utility functions and hooks, which lack proper documentation comments. These comments are useful because the IDE/text editor (which has proper `LSP` support) can parse them and show such documentation in hover popups, which adds a lot of value.

E.g. I am taking Sublime Text as an example with the `LSP` & `LSP-typescript` installed (Should be similar in VS Code)

Hovering over `lighten` (which does not have a documentation comment by default)

![image](https://user-images.githubusercontent.com/34304644/147811143-dbaa582a-c619-4b9e-926c-e23fff5499a7.png)

Hovering over `lighten` (which has a documentation comment)

![image](https://user-images.githubusercontent.com/34304644/147811162-fa1280e8-5d95-4d34-8800-6098fd55c26d.png)

**Comment by rtivital** [comment]
These functions are very basic, I think that variables name are descriptive enough

**Reply by UltraInstinct05** [reply]
It's really about the UX while using the library. A lot of people don't have the patience to sift through the codebase to understand how the function works. Taking `lighten` as an example, it takes a `color` string value. Can it be in any color format ? What are the limitations. These are some small stuff that can easily be clarified by inline docs IMHO. Anyways, I can't force someone to write doc comments.
